main/src/ca/uwaterloo/flix/runtime/channel/channel.flix

```diff
@@ -0,0 +1,75 @@
+def newId(): ##GlobalCounter & Impure = 
+    import new flix.runtime.GlobalCounter() as newGlobalCounter;
+    import flix.runtime.GlobalCounter:newId() as javaNewId;
+    javaNewId(newGlobalCounter())
+
+type alias Lock = ##java.util.concurrent.locks.Lock
+
+type alias ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
+
+def newReentrantLock(): ReentrantLock & Impure = 
+    import new java.util.concurrent.locks.ReentrantLock() as javaNewReentrantLock;
+    javaNewReentrantLock()
+
+type alias Condition = ##java.util.concurrent.locks.Condition
+
+def newCondition(reentrantLock: ReentrantLock): Condition & Impure =
+    import java.util.concurrent.locks.ReentrantLock.newCondition() as javaNewCondition;
+    javaNewCondition(reentrantLock)
+
+type alias LockConditionPair = {
+    lock: Lock,
+    condition: Condition
+}
+
+type alias Channel[a] = {
+    id: Int32,
+    channelLock: ReentrantLock,
+    bufferSize: Int32,
+    unbuffered: Bool,
+    elementQueue: Array[a],
+    waitingGetters: MutSet[LockConditionPair],
+    waitingSetters: Condition
+}
+
+def newChannel(bufferSize: Int32): Channel & Impure =
```

Comment:
Note that `Channel` should be `Channel[a]`.

Its probably a bug in the compiler if it does not complain about this.